internal/util/projutil/exec.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package projutil
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	homedir "github.com/mitchellh/go-homedir"
+)
+
+func ExecCmd(cmd *exec.Cmd) error {
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to exec %#v: %v", cmd.Args, err)
+	}
+	return nil
+}
+
+// GoBuildOptions configures "go build" and "go test".
+type GoBuildOptions struct {
```

Comment:
Not a major issue but for the purpose of readability, passing `GoBuildOptions` to `GoTest(opts)` doesn't seem right to me.
If the options for `GoTest()` and `GoBuild` are identical (I think `go build` flags are a subset of `go test`) then we should rename the struct to something like `GoCmdOptions` and generalize the options:
```Go
// GoCmdOptions configures "go build" and "go test".
type GoCmdOptions struct {
	// BinName is the name of the compiled binary, passed to -o.
	BinName string
	// Args are args passed to "go build/test", aside from "-o {bin_name}".
	Args []string
	// PackagePath is the path to the main(go build) or test(go test) packages.
	PackagePath string
	// Env is a list of environment variables to pass to the cmd;
	// exec.Command.Env is set to this value.
	Env []string
	// Dir is the dir to run "go build/test" in; exec.Command.Dir is set to this value.
	Dir string
	// GoMod determines whether to set the "-mod=vendor" flag.
	// If true, "go build" will use modules.
	// If false, "go build" will not use go modules. This is the default.
	GoMod bool
}
```

I have to read up on go modules but is `-mod=vendor` applicable for `go test`?

Reply:
I like the idea of having a more general options type name.

From `go help build`:

>The build flags are shared by the build, clean, get, install, list, run, and test commands

These build flags include `-mod`.